mlflow/pytorch/__init__.py

```diff
@@ -251,6 +439,7 @@ def save_model(
     input_example: ModelInputExample = None,
     requirements_file=None,
     extra_files=None,
+    save_as_state_dict=False,
```

Comment:
The code in save state dict is too clumsy to reuse. If we need to get rid of this boolean parameter and have an independent method as `save_stat_dict`, we have to split the `save_model` into three parts.
  **Where the condition checks are done and model path is derived**
https://github.com/mlflow/mlflow/blob/284742591ae3a7cd05ea03159ffa9d9ebe905163/mlflow/pytorch/__init__.py#L365-L401 

**Actual model saving part using torch.save**
https://github.com/mlflow/mlflow/blob/284742591ae3a7cd05ea03159ffa9d9ebe905163/mlflow/pytorch/__init__.py#L402-L405 



**post tasks like saving extra files, updating conda.yaml etc.**
https://github.com/mlflow/mlflow/blob/284742591ae3a7cd05ea03159ffa9d9ebe905163/mlflow/pytorch/__init__.py#L407-L467

Reply:
@harupy @chauhang Please let me know if we need to split the method and re-use the code or we can have the boolean variable action as above.